instana/instrumentation/aiohttp/server.py

```diff
@@ -37,7 +37,13 @@ async def stan_middleware(request, handler):
                     if custom_header in request.headers:
                         scope.span.set_tag("http.%s" % custom_header, request.headers[custom_header])
 
-            response = await handler(request)
+            response = None
+            try:
+                response = await handler(request)
+            except aiohttp.web.HTTPException as e:
+                # AIOHTTP uses exceptions for specific responses
+                # see https://docs.aiohttp.org/en/latest/web_exceptions.html#web-server-exceptions
+                response = e
```

Comment:
In AIOHTTP their exception objects have the status & headers attributes just like the response objects (same interface) so this shouldn't break the lines below. https://docs.aiohttp.org/en/latest/web_exceptions.html#base-http-exception